### PR TITLE
{numlib}[iimpi/2020.12] imkl v2021.1.1

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
requires ~~https://github.com/easybuilders/easybuild-framework/pull/3665~~ (iimpi toolchain fixes) + ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2407~~ (updated imkl easyblock) + https://github.com/easybuilders/easybuild-easyblocks/pull/2430 (make `imkl` easyblock unpack `examples`)

### CODE REVIEW

File: easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+homepage = 'https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onemkl.html'
+description = "Intel oneAPI Math Kernel Library"
+
+toolchain = {'name': 'iimpi', 'version': '2020.12'}
+
+# see https://software.intel.com/content/www/us/en/develop/articles/oneapi-standalone-components.html
+source_urls = ['https://registrationcenter-download.intel.com/akdlm/irc_nas/17402/']
+sources = ['l_onemkl_p_%(version)s.52_offline.sh']
+checksums = ['818b6bd9a6c116f4578cda3151da0612ec9c3ce8b2c8a64730d625ce5b13cc0c']
+
```
Comment (bartoldeman) [inline easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb:13]: I noticed that the `postinstallcmds` block with the examples has been removed. Did you intend to move that to the easyblock?

**Review by bartoldeman — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name = 'imkl'
+version = '2021.1.1'
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb:2]:
```suggestion
version = '2021.2.0'
```

File: easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+homepage = 'https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onemkl.html'
+description = "Intel oneAPI Math Kernel Library"
+
+toolchain = {'name': 'iimpi', 'version': '2020.12'}
+
+# see https://software.intel.com/content/www/us/en/develop/articles/oneapi-standalone-components.html
+source_urls = ['https://registrationcenter-download.intel.com/akdlm/irc_nas/17402/']
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb:10]:
```suggestion
source_urls = ['https://registrationcenter-download.intel.com/akdlm/irc_nas/17757']
```

File: easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+sources = ['l_onemkl_p_%(version)s.52_offline.sh']
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb:11]:
```suggestion
sources = ['l_onemkl_p_%(version)s.296_offline.sh']
```

File: easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+checksums = ['818b6bd9a6c116f4578cda3151da0612ec9c3ce8b2c8a64730d625ce5b13cc0c']
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb:12]:
```suggestion
checksums = ['816e9df26ff331d6c0751b86ed5f7d243f9f172e76f14e83b32bf4d1d619dbae']
```

File: easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+name = 'imkl'
+version = '2021.1.1'
+
+homepage = 'https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onemkl.html'
+description = "Intel oneAPI Math Kernel Library"
+
+toolchain = {'name': 'iimpi', 'version': '2020.12'}
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb:7]:
```suggestion
toolchain = {'name': 'iimpi', 'version': '2021.03'}
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]
Do we want to bump this to the latest imkl v2021.2.0?

**Comment by boegel** [comment]
@SebastianAchilles We'll use imkl 2021.2 in `intel/2021a` indeed, but I think merging this older `imkl` 2021.1 version is worth it too. I'll open a separate PR for `imkl` 2021.2 once this one is merged.

File: easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/i/imkl/imkl-2021.1.1-iimpi-2020.12.eb:13]: Yeah, but then I forgot about it. :D

That's done now in https://github.com/easybuilders/easybuild-easyblocks/pull/2430 (those changes will still work with existing `imkl` easyconfigs that still include the `postinstallcmds`, but we should clean those up).

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 2430"

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2430
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bber0501u03a.bb2.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/0e548ec10722996e830cca9981ce19f2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2430
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3506.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/905a87d4509e4ba90afa3d3c4613023a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2430
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2618.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/eb54062281c0123b42d60c17b4e0565f for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2430
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/c4581d2dab07154104eb2f5d0ed11b59 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2430
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/27e4822ad7bb202dc34d9dd9a1414452 for a full test report.

**Comment by boegel** [comment]
@branfosj Good to go, closely followed by https://github.com/easybuilders/easybuild-easyconfigs/pull/12876?

**Review by branfosj — APPROVED** [review]
lgtm

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Pull request merged**